src/sonic-frr/patch/0037-bgpd-Ensure-community-data-is-freed-in-some-cases.patch

```diff
@@ -0,0 +1,246 @@
+ 				     struct attr *attr,
+ 				     struct bgp_path_info *path);
+diff --git a/bgpd/bgp_updgrp_adv.c b/bgpd/bgp_updgrp_adv.c
+index af8ef751da..22af326230 100644
+--- a/bgpd/bgp_updgrp_adv.c
++++ b/bgpd/bgp_updgrp_adv.c
+@@ -454,7 +454,7 @@ bgp_advertise_clean_subgroup(struct update_subgroup *subgrp,
+ 	return next;
+ }
+ 
+-void bgp_adj_out_set_subgroup(struct bgp_dest *dest,
++bool bgp_adj_out_set_subgroup(struct bgp_dest *dest,
+ 			      struct update_subgroup *subgrp, struct attr *attr,
+ 			      struct bgp_path_info *path)
+ {
+@@ -474,7 +474,7 @@ void bgp_adj_out_set_subgroup(struct bgp_dest *dest,
+ 	bgp = SUBGRP_INST(subgrp);
+ 
+ 	if (DISABLE_BGP_ANNOUNCE)
+-		return;
++		return false;
+ 
+ 	/* Look for adjacency information. */
+ 	adj = adj_lookup(
+@@ -490,7 +490,7 @@ void bgp_adj_out_set_subgroup(struct bgp_dest *dest,
+ 			bgp_addpath_id_for_peer(peer, afi, safi,
+ 						&path->tx_addpath));
+ 		if (!adj)
+-			return;
++			return false;
+ 
+ 		subgrp->pscount++;
+ 	}
+@@ -529,7 +529,7 @@ void bgp_adj_out_set_subgroup(struct bgp_dest *dest,
+ 		 * will never be able to coalesce the 3rd peer down
+ 		 */
+ 		subgrp->version = MAX(subgrp->version, dest->version);
+-		return;
++		return false;
+ 	}
+ 
+ 	if (adj->adv)
+@@ -576,6 +576,8 @@ void bgp_adj_out_set_subgroup(struct bgp_dest *dest,
+ 	bgp_adv_fifo_add_tail(&subgrp->sync->update, adv);
+ 
+ 	subgrp->version = MAX(subgrp->version, dest->version);
++
++	return true;
+ }
+ 
+ /* The only time 'withdraw' will be false is if we are sending
+@@ -668,7 +670,7 @@ void subgroup_announce_table(struct update_subgroup *subgrp,
+ {
+ 	struct bgp_dest *dest;
+ 	struct bgp_path_info *ri;
+-	struct attr attr;
++	struct attr attr = {0};
+ 	struct peer *peer;
+ 	afi_t afi;
+ 	safi_t safi;
+@@ -715,17 +717,20 @@ void subgroup_announce_table(struct update_subgroup *subgrp,
+ 						    &attr, NULL)) {
+ 				/* Check if route can be advertised */
+ 				if (advertise) {
+-					if (!bgp_check_withdrawal(bgp, dest))
+-						bgp_adj_out_set_subgroup(
++					if (!bgp_check_withdrawal(bgp, dest)) {
++						if (!bgp_adj_out_set_subgroup(
+ 							dest, subgrp, &attr,
+-							ri);
+-					else
++							ri))
++							bgp_attr_flush(&attr);
++					} else {
+ 						bgp_adj_out_unset_subgroup(
+ 							dest, subgrp, 1,
+ 							bgp_addpath_id_for_peer(
+ 								peer, afi,
+ 								safi_rib,
+ 								&ri->tx_addpath));
++						bgp_attr_flush(&attr);
++					}
+ 				}
+ 			} else {
+ 				/* If default originate is enabled for
+@@ -745,6 +750,7 @@ void subgroup_announce_table(struct update_subgroup *subgrp,
+ 					bgp_addpath_id_for_peer(
+ 						peer, afi, safi_rib,
+ 						&ri->tx_addpath));
++				bgp_attr_flush(&attr);
+ 			}
+ 		}
+ 	}
+@@ -947,7 +953,7 @@ void subgroup_default_originate(struct update_subgroup *subgrp, int withdraw)
+ 		if (dest) {
+ 			for (pi = bgp_dest_get_bgp_path_info(dest); pi;
+ 			     pi = pi->next) {
+-				if (CHECK_FLAG(pi->flags, BGP_PATH_SELECTED))
```

Comment:
Can you please check the original PR for this file as well? The diff is little bit different

Reply:
> Can you please check the original PR for this file as well? The diff is little bit different

Yes, this diff was based on our understanding. we will sync with donaldsharp first, and push the latest update if necessary later, thanks.